internal/ingress/controller/template/template.go

```diff
@@ -1672,3 +1673,42 @@ func convertGoSliceIntoLuaTable(goSliceInterface interface{}, emptyStringAsNil b
 		return "", fmt.Errorf("could not process type: %s", kind)
 	}
 }
+
+// buildOriginRegex returns an origin as a regex
+func buildOriginRegex(origin string) string {
+	if !strings.HasPrefix(origin, "*") {
+		return origin
+	}
```

Comment:
This seems to be saying that if it doesn't start with a `*`, then it's not going to escape the regex, so `foo.bar.com` will match `fooxbar.com`?

Reply:
yes, you are right to seem pretty open. IMHO, below are the two solutions wd be perfect
1. there has to be a way to enforce user-defined policy (e.g. nginx.ingress.kubernetes.io/cors-allow-origin-policy: "user-defined regex") to match "http_origin"
2. User should list all the subdomains (foo.bar.com, foo1.bar.com, yar.bar.com, bkar.foo.bar.com ......) in NginX ConfigMap for engress to exact match.

Reply:
> This seems to be saying that if it doesn't start with a `*`, then it's not going to escape the regex, so `foo.bar.com` will match `fooxbar.com`?

I'll test this however, because of the '|' I would expect an exact match

Reply:
`|` means match the regular expression on the left or the regular expression on the right, it's definitely not an exact match:

https://play.golang.org/p/RxPnmuQ19cj

Reply:
Alright, will check this

Reply:
> 
> 
> yes, you are right to seem pretty open. IMHO, below are the two solutions wd be perfect
> 
>     1. there has to be a way to enforce user-defined policy (e.g. nginx.ingress.kubernetes.io/cors-allow-origin-policy: "user-defined regex") to match "http_origin"
> 
>     2. User should list all the subdomains (foo.bar.com, foo1.bar.com, yar.bar.com, bkar.foo.bar.com ......) in NginX ConfigMap for ingress to exact match.

this is a good idea, imo. to provide filtering based on what the user wants instead of hacking away.

Reply:
> 
> 
> yes, you are right to seem pretty open. IMHO, below are the two solutions wd be perfect
> 
>     1. there has to be a way to enforce user-defined policy (e.g. nginx.ingress.kubernetes.io/cors-allow-origin-policy: "user-defined regex") to match "http_origin"
> 
>     2. User should list all the subdomains (foo.bar.com, foo1.bar.com, yar.bar.com, bkar.foo.bar.com ......) in NginX ConfigMap for ingress to exact match.

Currently, this is not what is done but maybe you should raise an issue with this feature.
It could probably be selected for another milestone perhaps? This is currently not in the PR.